packages/contracts/contracts/GenArt721CoreV4.sol

```diff
@@ -0,0 +1,2010 @@
+    ) ERC721_PackedHashSeed(_tokenName, _tokenSymbol) {
+        _onlyNonZeroAddress(_randomizerContract);
+        // record contracts starting project ID
+        // casting-up is safe
+        startingProjectId = uint256(_startingProjectId);
+        _updateArtblocksPrimarySalesAddress(msg.sender);
+        _updateArtblocksSecondarySalesAddress(msg.sender);
+        _updateRandomizerAddress(_randomizerContract);
+        // set AdminACL management contract as owner
+        _transferOwnership(_adminACLContract);
+        // initialize default base URI
+        _updateDefaultBaseURI("https://token.artblocks.io/");
+        // initialize next project ID
+        _nextProjectId = _startingProjectId;
+        emit PlatformUpdated(FIELD_NEXT_PROJECT_ID);
+    }
+
+    /**
+     * @notice Mints a token from project `_projectId` and sets the
+     * token's owner to `_to`. Hash may or may not be assigned to the token
+     * during the mint transaction, depending on the randomizer contract.
+     * @param _to Address to be the minted token's owner.
+     * @param _projectId Project ID to mint a token on.
+     * @param _by Purchaser of minted token.
+     * @return _tokenId The ID of the minted token.
+     * @dev sender must be the allowed minterContract
+     * @dev name of function is optimized for gas usage
+     */
+    function mint_Ecf(
+        address _to,
+        uint256 _projectId,
+        address _by
+    ) external returns (uint256 _tokenId) {
+        // CHECKS
+        require(msg.sender == minterContract, "Must mint from minter contract");
+        Project storage project = projects[_projectId];
+        // load invocations into memory
+        uint24 invocationsBefore = project.invocations;
+        uint24 invocationsAfter;
+        unchecked {
+            // invocationsBefore guaranteed <= maxInvocations <= 1_000_000,
+            // 1_000_000 << max uint24, so no possible overflow
+            invocationsAfter = invocationsBefore + 1;
+        }
+        uint24 maxInvocations = project.maxInvocations;
+
+        require(
+            invocationsBefore < maxInvocations,
+            "Must not exceed max invocations"
+        );
+        require(
+            project.active ||
+                _by == projectIdToFinancials[_projectId].artistAddress,
+            "Project must exist and be active"
+        );
+        require(
+            !project.paused ||
+                _by == projectIdToFinancials[_projectId].artistAddress,
+            "Purchases are paused."
+        );
+
+        // EFFECTS
+        // increment project's invocations
+        project.invocations = invocationsAfter;
+
+        // mark project as completed if hit max invocations
+        if (invocationsAfter == maxInvocations) {
+            _completeProject(_projectId);
+        }
+
+        // Mint to the artist's contract.
+        uint256 thisTokenId = IManifoldCore(project.artistContractAddress)
+            .mintExtension(_to);
```

Comment:
This is where the token actually gets minted now (instead of to the GenArt contract)

Reply:
This wouldn't work because if you have multiple `artistContracts` giving a `tokenId = 1`, then they'll keep overwriting each other.

Reply:
Great catch, because I am saving it below -

```
tokenIdToProjectIds[thisTokenId] = _projectId;
```

Will really need to have that be like this instead - 

```
tokenIdToProjectIds[project.artistContractAddress][thisTokenId] = _projectId;
```

Will update!